accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -46,214 +46,205 @@ extension MyFancyList<T> on List<T> {
 
-If there is no such member, the operation is currently a compile-time error. In that case, all extensions in scope are checked for whether they apply. *That is, extension members only apply to code that would currently be a compile-time error*. An extension applies to a member access if the static type of the receiver is a subtype of the `on` type of the extension *and* the extension has an instance member with the same base name as the operation. 
+This means that you cannot do cascade invocations on explicit extension applications: `E(e)..foo()..bar()` is a compile-time error. This is necessary because that expression evaluates to the value of `E(e)`, and an extension application does not have a value.
 
-For generic extensions, standard type inference is used to infer the type arguments before comparing to the `on` type. As an example, take the following extension:
+if *A* is a member invocation with an extension application of an extension *E* as target expression, then type inference applies to the member invocation. If `E` is declared as
 
 ```dart
-extension MyList<T extends Comparable<T>> on List<T> {
-  void quickSort() { ... }
+extension E<X...> on T {
+  ... members ...
 }
 ```
 
-and a member access like:
+then the type inference on *A* is the same that would be applied to the member invocation on $E$ considered as a constructor invocation on a class declared as:
 
 ```dart
-List<Duration> times = ...;
-times.quickSort();
+class E<X...> {
+  final T $target;
+  E(this.$target);
+  ... members // with inference applied to the body, including implicit extension 
+              // member invocations as described in later sections, 
+              // and with `$target` instead of `this` ...
+}
 ```
 
-Here we perform type inference equivalent to what we would do for:
+That is, if `E` declares an instance member `T foo(T arg)`, then the inference of `E(e1).foo(e2)` will first perform inference to `E(e1)` as described above, and then perform inference on the member invocation just as if it was a class member. It is a **compile-time error** if this class member invocation would be a compile-time error
 
-```dart
-class MyList<T extends Comparable<T>> {
-  MyList(List<T> argument);
-  void quickSort() { ... }
-}
-...
-List<Duration> times = ...;
-... MyList(times).quickSort() ... // Infer type argument of MyList here.
-```
+The static type of a member invocation on an extension application is the return type of the extension member with the corresponding member name of the invocation, with the explicit or inferred type arguments of the extension application replacing the type parameters.
```

Comment:
"type parameters bound by the extension, and the explicit or inferred type arguments of the invoked member replacing the type parameters bound by the member".